### PR TITLE
Malaria Detection

### LINKED ISSUE #893: Addition of Malaria detection from cell image
### **Aim**
Develop a web app that predicts malaria based on input of cell image. It will be a web page with flask framework as backend which will work on deep learning model of malaria trained image.

I'll like to work , if assigned.

### PR DESCRIPTION
**Pull request by ayush-raj8**
# Description
Python script and model to detect malaria infected cells.

## Fixes #893

Replace `893` with the issue number which is fixed in this PR 927

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [x] Yes

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [x] My changes generate no new warnings

### CODE REVIEW

File: Malaria/Dataset/Dataset Link.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
```
Comment (kaustubhgupta) [inline Malaria/Dataset/Dataset Link.txt:1]: this file is not needed , you can delete it

File: Malaria/main.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+from PIL import Image
+import tensorflow as tf
+import numpy as np
+
+# Configuring user's pc so that tensorflow model built on other's pc can run
+os.environ['TF_CPP_MIN_LOG_LEVEL'] = '2'
+os.environ['TF_XLA_FLAGS'] = '--tf_xla_enable_xla_devices'
+
+# reading image
+img = Image.open("Images/cell1.jpg")
```
Comment (kaustubhgupta) [inline Malaria/main.py:13]: Take input for image path from user

File: Malaria/main.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+# confiiguring input image according to the model's requirement
+img = img.resize((36, 36))
+img = np.asarray(img)
+img = img.reshape((1, 36, 36, 3))
+img = img.astype(np.float64)
+model_path = "Models/malaria.h5"
```
Comment (kaustubhgupta) [inline Malaria/main.py:19]:
```suggestion
model_path = "./Malaria/Models/malaria.h5"
```

File: Malaria/Model development/malariadetection.ipynb [hunk 4]
```diff
@@ -0,0 +1,450 @@
+{
```
Comment (kaustubhgupta) [inline Malaria/Model development/malariadetection.ipynb:1]: Convert this notebook into .py script and delete this notebook

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

File: Malaria/script.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
+# importing the libraries for loading data and visualisation
```
Comment (kaustubhgupta) [inline Malaria/script.py:1]: name this file as `model_training.py` instead of `script.py`

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Delete the malaraidetection.ipynb file along with this change:

File: Malaria/main.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+from PIL import Image
+import tensorflow as tf
+import numpy as np
+
+# Configuring user's pc so that tensorflow model built on other's pc can run
+os.environ['TF_CPP_MIN_LOG_LEVEL'] = '2'
+os.environ['TF_XLA_FLAGS'] = '--tf_xla_enable_xla_devices'
+
+# reading image
+val_path = input("Enter image path name eg. Images/cell1.jpg")
```
Comment (kaustubhgupta) [inline Malaria/main.py:13]:
```suggestion
val_path = input("Enter image path name eg. Images/cell1.jpg: ")
```

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
For the third time: DELETE THIS FILE! malariadetection.ipynb

**Comment by ayush-raj8** [comment]
@kaustubhgupta  This new deep source python is showing errors because of comments and line too long.  This code runs perfectly fine on any python ide e.g. Pycharm.
Line too long cannot be removed as it involves model formation.
image
image

Lines will have more than 88 characters , because this is the standard way of training , testing and forming CNN.

**Review by kaustubhgupta — APPROVED** [review]

**Review by Kushal997-das — APPROVED** [review]
LGTM 👍

**Review by santushtisharma10 — APPROVED** [review]

**Comment by antrikshmisri** [comment]
> @kaustubhgupta  This new deep source python is showing errors because of comments and line too long.  This code runs perfectly fine on any python ide e.g. Pycharm.
> Line too long cannot be removed as it involves model formation.
> image
> image
> 
> Lines will have more than 88 characters , because this is the standard way of training , testing and forming CNN.
> 

@ayush-raj8 Have a look at pep8 guidelines for long lines [here](https://www.python.org/dev/peps/pep-0008/#:~:text=The%20preferred%20way%20of%20wrapping,a%20backslash%20for%20line%20continuation)

**Review by antrikshmisri — APPROVED** [review]

**Review by avinashkranjan — APPROVED** [review]
LGTM

**Pull request merged**